subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt.c

```diff
@@ -610,7 +610,11 @@ img_mgmt_my_version(struct image_version *ver)
 static const struct mgmt_handler img_mgmt_handlers[] = {
 	[IMG_MGMT_ID_STATE] = {
 		.mh_read = img_mgmt_state_read,
+#ifdef CONFIG_MCUBOOT_BOOTLOADER_USES_DIRECT_XIP
+		.mh_write = NULL
+#else
```

Comment:
We have 2 possible ways of doing this, one with Kconfig here or another whereby it is always enabled but at runtime works out which mode it is in: https://github.com/zephyrproject-rtos/zephyr/pull/55805 (or, a third way whereby it does both, always at runtime and make the Kconfig optional which, if enabled, then forces a particular code path) - thoughts?

Reply:
Generally it could work, currently makes no sense as other parts of MCUmgr are now hardcoded to either work as DirectXIP or as swap.
This would make a lot of sense if we would figure out how to build app in slot independent way. Then we could have the same app running as DirectXIP and as swap, in which case it **would be mandatory to** check MCUboot config.
Btw. would it be possible to actually share some callback vector via retention API so that MCUboot would provide the confirmation/test API?

Reply:
> Btw. would it be possible to actually share some callback vector via retention API so that MCUboot would provide the confirmation/test API?

It could, but that then introduces other problems, e.g. for security you might want to prevent reading of the mcuboot code from the application (ncs does this) in which case you could share it but calling it would fault the CPU. There's also privilege level of the callee which might do the same. There is also the potential problem that mcuboot cannot be updated so if there is a security flaw in that code, anything could call it and it cannot be patched.